### DISCUSSION TITLE
How is this different from aniseed?

### DISCUSSION THREAD

**Discussion by glyh** [Olical/nfnl#25] [Q&A] [answered]
As title

**Comment by Olical** [comment, marked as answer]
So I think this answer sort of answers things https://github.com/Olical/nfnl/discussions/1#discussioncomment-6386694

But the highlights are:

 - No special syntax, just normal Fennel that doesn't _need_ nfnl, it just helps you compile it.
 - Only loads when you're modifying a Fennel file that has a `.nfnl.fnl` config file nearby, will never load otherwise.
 - Compiles at file modification time meaning there's NO startup checking etc to see if things need to compile.
 - Works on non-Neovim related things, since it's really a Fennel compiler hooked into Neovim but nothing more you can use it to work on non-Neovim Fennel projects.
 - Works in your home dir in the same way as a plugin in the same way as a LOVE game, they're all the same now. It leans on native Neovim things at all points so it does the least possible work to help integrate Fennel then steps out of the way.
 - Works with macros properly! You just have to mark macro files with a comment but I think that's a small price to pay for otherwise pretty flawless integration (I think?). Was always a pain with Aniseed.
 - Because it leans on native Neovim stuff, there's far less code, most of the things I was re-inventing is now just native Neovim+Lua stuff.
 - The config file based approach means your plugin can have some config and your dotfiles can have something else. Any other person modifying your dotfiles or plugin will also have their nfnl read that config and use it. Creating a standardised config system that is shared between users per repo (or even directory).
 - Auto generated docs! Because the Fennel is vanilla you can generate documentation, which I do inside this repo.
 - LSP / linting works! No more Aniseed macros means you get Fennel linting like unused requires etc.
 - You still have access to the `autoload` system and `aniseed.core` but under `nfnl.core`, same thing ported over. We could even port the module macros over, you'd just need to explicitly require them in your code to use them.

All in all, I just did the same project again but rewrote the fundamental decisions I regretted, learned from a bunch of painful lessons and leaned on features of Neovim that didn't exist when I wrote Aniseed.

I hope this helps put it in perspective!

**Reply by Olical** [reply]
And because of the config file approach with NO global config it means we could start specifying Fennel compiler versions per project... I haven't done that yet, but that's a good example of the sort of flexibility we have with this project in comparison. The global config massively hindered Aniseed. I'd even take a similar approach with Conjure these days I think.
